web/src/app/services/ServiceLabelEditDialog.js

```diff
@@ -26,56 +26,63 @@ const query = gql`
   }
 `
 
-export default class ServiceLabelCreateDialog extends React.PureComponent {
-  static propTypes = {
-    serviceID: p.string.isRequired,
-    labelKey: p.string.isRequired,
-    onClose: p.func,
-  }
-
-  state = {
-    value: null,
-    errors: [],
-  }
-
-  render() {
-    return this.renderQuery()
-  }
+export default function ServiceLabelCreateDialog(props) {
+  const [value, setValue] = useState(null)
+  const { onClose, labelKey, serviceID } = props
```

Comment:
nit, swap these two lines to keep prop destructering at the top:
```suggestion
  const { onClose, labelKey, serviceID } = props
  const [value, setValue] = useState(null)
```